rafx-base/src/slab/drop_slab.rs

```diff
@@ -126,6 +58,14 @@ impl<T> DropSlab<T> {
         DropSlabKey::new(slab_key, self.drop_tx.clone())
     }
 
+    pub fn allocate_raw(
```

Comment:
What is this used for? Do we need one slab implementation to support both? (Wondering if it's cleaner to have two separate slab types.)

Reply:
A `RenderObjectHandle` is a reference-counted handle for a `RenderObject` associated with some `RenderFeature`. The `RenderObject`s are allocated out of some `DropSlab`. We'd like the `RenderObjectHandle` to contain the key so that 1.) it can be used for the look-up, and 2.) when there are no more references to the `RenderObjectHandle`, the `DropSlab` can free that slot. However, we also want to put `RenderObjectHandle`s into lists or hash maps or other containers. If the `DropSlabKey` includes the generic `T`, we can't do that because the different `T` arguments will bleed into the `RenderObjectHandle`. 

So, this allocates a key that is _not_ generic on `T`. Other than that it's the exact same as `DropSlabKey`. If you think back to the code today, it's basically the same as the `GenericNodeRenderHandle` if the `GenericNodeRenderHandle` is what truly represented "ownership" of the entry in the slab that allocated nodes.

I would maybe lean towards nixing the `T` from `DropSlabKey` and then getting rid of the key type that I added? If you're concerned about keys allocated from one `T` being used in a different `T`, we could include the `TypeId` and assert on that in the various `get` functions, or do some type of equality check using the `Sender` on the key (assuming we can compare it against our `Receiver`?)

Reply:
Removed `T` from `DropSlabKey`. When querying with the key, use `same_channel` to check that the key is from the right `DropSlab`.

Reply:
I see, if we need to have typed/untyped handles, I would have the API always use typed handles, but allow converting typed -> untyped and untyped -> typed. If it's painful to do the explicit conversion back to typed (because it requires generics where we don't want them) then just asserting on typeid is fine.